ignite/handlers/checkpoint.py

```diff
@@ -253,19 +256,52 @@ def last_checkpoint(self) -> Optional[str]:
         return self._saved[-1].filename
 
     def _check_lt_n_saved(self, or_equal=False):
-        if self._n_saved is None:
+        if self.n_saved is None:
             return True
-        return len(self._saved) < self._n_saved + int(or_equal)
+        return len(self._saved) < self.n_saved + int(or_equal)
+
+    def _add_field(self, f, sep="_"):
+        return "{}{}".format(f, sep) if len(f) > 0 else f
+
+    def _get_filename_pattern(
```

Comment:
Here, arguments `Callable` are used as a boolean flag. Why do not introduce
```
def _get_filename_pattern(
    self,
    filename_prefix: str = "",
    has_score_fn: book = False,
    has_score: bool = False,
    ...
)
```
What do you think?

Reply:
@sdesrozis it seems like a good choice. Are you preferring to do the following when calling `_get_filename_pattern`?
```
filename_pattern = self._get_filename_pattern(
                filename_prefix=self.filename_prefix,
                has_score_fn=True if self.score_function is None else False,
                has_score_name=True if self.score_name is None else False,
                has_global_step_transform=True if self.global_step_transform is None else False,
                filename_pattern=self.filename_pattern,
            )
```

Reply:
@sdesrozis was busy with some work. Is the above one ok shall I make this change?

Reply:
@Joel-hanson maybe it could make sense to make this function public and static... 
`_get_filename_pattern` -> `setup_filename_pattern`

Reply:
Even more simple: 
```
filename_pattern = Checkpoint.setup_filename_pattern(
                filename_prefix=self.filename_prefix,
                has_score_fn=self.score_function is None,
                has_score_name=self.score_name is None,
                has_global_step_transform=self.global_step_transform is None,
            )
```

Reply:
Will update accordingly @vfdev-5

Reply:
@vfdev-5 will the naming be confusing when having `with_score=self.score_function is None` 
```
if with_score and with_score_name and with_global_step_transform:
    filename_pattern = "{name}_{score}.{ext}"
```
is it ok to have `with_score=self.score_function is not None` , this applies to others variables as well

Reply:
I think we can explain the meaning of `with_score` in docstring of the method